src/pages/home/report/ReportActionsView.tsx

```diff
@@ -233,12 +343,68 @@ function ReportActionsView({
         } else {
             Performance.markEnd(CONST.TIMING.SWITCH_REPORT);
         }
-    }, [hasCachedActions]);
+    }, [hasCachedActionOnFirstRender]);
+
+    useEffect(() => {
+        // Temporary solution for handling REPORTPREVIEW. More details: https://expensify.slack.com/archives/C035J5C9FAP/p1705417778466539?thread_ts=1705035404.136629&cid=C035J5C9FAP
+        // This code should be removed once REPORTPREVIEW is no longer repositioned.
+        // We need to call openReport for gaps created by moving REPORTPREVIEW, which causes mismatches in previousReportActionID and reportActionID of adjacent reportActions. The server returns the correct sequence, allowing us to overwrite incorrect data with the correct one.
+        const shouldOpenReport =
+            newestReportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW &&
+            !hasCreatedAction &&
+            isReadyForCommentLinking &&
+            reportActions.length < 24 &&
+            reportActions.length >= 1 &&
+            !isLoadingInitialReportActions &&
+            !isLoadingOlderReportActions &&
+            !isLoadingNewerReportActions;
+
+        if (shouldOpenReport) {
+            Report.openReport(reportID, reportActionID);
+        }
+    }, [
+        hasCreatedAction,
+        reportID,
+        reportActions,
+        reportActionID,
+        newestReportAction?.actionName,
+        isReadyForCommentLinking,
+        isLoadingOlderReportActions,
+        isLoadingNewerReportActions,
+        isLoadingInitialReportActions,
+    ]);
+
+    // Check if the first report action in the list is the one we're currently linked to
+    const isTheFirstReportActionIsLinked = newestReportAction?.reportActionID === reportActionID;
+
+    useEffect(() => {
+        let timerID: NodeJS.Timeout;
+
+        if (isTheFirstReportActionIsLinked) {
+            setNavigatingToLinkedMessage(true);
+        } else {
+            // After navigating to the linked reportAction, apply this to correctly set
+            // `autoscrollToTopThreshold` prop when linking to a specific reportAction.
+            InteractionManager.runAfterInteractions(() => {
+                // Using a short delay to ensure the view is updated after interactions
+                timerID = setTimeout(() => setNavigatingToLinkedMessage(false), 10);
+            });
+        }
+
+        return () => {
+            if (!timerID) {
+                return;
+            }
+            clearTimeout(timerID);
+        };
+    }, [isTheFirstReportActionIsLinked]);
 
     // Comments have not loaded at all yet do nothing
     if (!reportActions.length) {
         return null;
     }
+    // AutoScroll is disabled when we do linking to a specific reportAction
+    const shouldEnableAutoScroll = hasNewestReportAction && (!reportActionID || !isNavigatingToLinkedMessage);
```

Comment:
We should also auto scroll if we switch from many-transactions-view to one-transaction-view (Coming from https://github.com/Expensify/App/issues/47584)